### PR TITLE
Feature/412 spc reader

### LINKED ISSUE #412: .spc reader from SPCinspect to pytrip?
https://github.com/pytrip/SPCinspector/blob/master/spcinspect/spcread.py

### PR DESCRIPTION
**Pull request by grzanka**
Fixes #412

### CODE REVIEW

**Comment by grzanka** [comment]
@nbassler it seems working - can you review ?

File: pytrip/spc.py [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+            self.peakpos = payload
+            return payload
+
+        if tag.code == 8:
+            sdtype = ste + 'f' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0]
+            self.norm = payload
+            return payload
+
+        if tag.code == 9:  # number of depth steps
```
Comment (nbassler) [inline pytrip/spc.py]: Just a note: I think this will probably be set to unity for SPC applied to a single voxel.

File: pytrip/spc.py [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+                fd.write(np.asarray(specie.rcumdata, dtype="{:s}f8".format(tag._ste)))
+
+        fd.close()
+
+    def read_data(self):
+        fd = open(self.filename, "rb")
+        t = Tag(fd)
+        db = []
+
+        while t.code < 9:
```
Comment (nbassler) [inline pytrip/spc.py]: how about using all caps named var for this to make it more readable?
t.code < SPC_TAG_FOOBAR:

File: pytrip/spc.py [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+
+        if tag.code == 16:  # nE: number of energy bins
+            sdtype = ste + 'u' + str(tag.size)
+            payload = np.fromfile(fd, count=cnt, dtype=np.dtype(sdtype))[0]
+            # self.ne = payload
+            return payload
+
+        if tag.code == 17:  # E: energy bin values
+            sdtype = ste + 'f8'
+            cnt = int(tag.size / 8)
```
Comment (nbassler) [inline pytrip/spc.py]: how about `cnt = tag.size // 8`

File: pytrip/spc.py (hunk 2 above)
Comment (nbassler) [reply pytrip/spc.py]: using a dict of tags

Comment (grzanka) [reply pytrip/spc.py]: See proposed solution here 19aa2c1

File: pytrip/spc.py (hunk 3 above)
Comment (grzanka) [reply pytrip/spc.py]: Done in 19aa2c1

File: pytrip/spc.py (hunk 1 above)
Comment (grzanka) [reply pytrip/spc.py]: You mean number of steps ?

**Pull request merged**